src/main/scala/io/scalechain/blockchain/BlockDataInputStream.scala

```diff
@@ -1,18 +1,42 @@
 package io.scalechain.blockchain
 
 import java.io.{InputStream, DataInputStream}
+import collection.mutable.HashMap
 
 import io.scalechain.blockchain.util.Utils
 
+class FieldStatsCalculator {
+  val fieldMap = new HashMap[String, Long]()
```

Comment:
The branch is for experimental purpose to calculate statistics of each field in blockchain. It runs in a single thread as well. So using a mutable map should be fine.